### ISSUE TITLE
User comments

### ISSUE THREAD

**Issue by anaisconce** [gristlabs/grist-core#236]
Make it possible for users to leave comments on cells and create discussion threads about your data.

**Comment by Alic94** [comment]
Is there any progress on the feature yet? 😄

**Comment by paulfitz** [comment]
Yes! Thanks to @berhalak. If you set a `COMMENTS` environment variable to `true`, the right-click menu for cells will have an option to add a comment, and a sidebar to see discussion threads. In using comments ourselves, we realize that it needs integration with a notification feature to be broadly useful, so we're not officially releasing the feature until we sort that out (no promises on timeline for that).

![Screenshot from 2023-01-23 11-45-02](https://user-images.githubusercontent.com/118367/214098393-73e701e2-a020-47c8-933a-dd4894df2031.png)

**Comment by Alic94** [comment]
Awesome! Thank you for the good news 😄

**Comment by toddz262** [comment]
I'm on the hosted free trial so I can't set environment variables, but FWIW I'd love to see this functionality sorted and launched. Record-level comments/conversation is pretty key in an Airtable setup I'd like to migrate from.

**Comment by paulfitz** [comment]
> I'm on the hosted free trial so I can't set environment variables, but FWIW I'd love to see this functionality sorted and launched. Record-level comments/conversation is pretty key in an Airtable setup I'd like to migrate from.

Are notifications a big deal for you? That's the main reason we're holding off, we don't have a good notifications solution (yet).

**Comment by toddz262** [comment]
@paulfitz - Thanks for asking. Notifications would be great eventually, but I imagine I could use a view based on recent timestamps as a workaround in the meantime.

**Comment by gest-ctrl** [comment]
> @paulfitz - Thanks for asking. Notifications would be great eventually, but I imagine I could use a view based on recent timestamps as a workaround in the meantime.

I'd love to have the comments functionality, and notifications are not a big deal to me.

**Comment by almereyda** [comment]
Then you can use comments today on a self-hosted instance. Follow these instructions above:

> If you set a `COMMENTS` environment variable to `true`, the right-click menu for cells will have an option to add a comment, and a sidebar to see discussion threads.

It gives you a new menu item:

![grafik](https://github.com/user-attachments/assets/5d06ad80-6af9-4037-b50c-0b94b9271666)

A new comment item in the context menu and a keyboard shortcut:

![grafik](https://github.com/user-attachments/assets/b8da4be8-8fb9-4047-be99-0b078befcbb4)

And a comments sidebar:

![grafik](https://github.com/user-attachments/assets/94e65d89-5dae-4059-980f-68ec246caf24)

The feature currently lacks a notification facility, yet is otherwise fully functional.

**Comment by tristanrobert** [comment]
> > I'm on the hosted free trial so I can't set environment variables, but FWIW I'd love to see this functionality sorted and launched. Record-level comments/conversation is pretty key in an Airtable setup I'd like to migrate from.
> 
> Are notifications a big deal for you? That's the main reason we're holding off, we don't have a good notifications solution (yet).

We are interested too by this feature: notification to users.

**Comment by dsagal** [comment]
Thanks to more work by @berhalak and others, Comments got more features (@-mentions of collaborators) and have been officially released.
